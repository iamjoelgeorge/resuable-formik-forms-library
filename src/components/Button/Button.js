import React from 'react';

import styles from './Button.module.scss';

const Button = (props) => {
  const { label, type, onClick, ...rest } = props;

  return (
    <button className={styles.button} type={type} {...rest}>
      Submit
    </button>
  );
};

export default Button;
