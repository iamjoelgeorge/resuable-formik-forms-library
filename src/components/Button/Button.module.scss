@import '../../theme/variables.scss';

.button {
  border: none;
  padding: 10px;
  border-radius: 2px;
  background-color: $darkPurple;
  color: white;
  font-weight: bold;
  margin-top: 12px;

  &:disabled {
    background-color: rgb(89, 89, 89);
  }
}
