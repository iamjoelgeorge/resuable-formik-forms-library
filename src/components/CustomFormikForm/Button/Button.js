import React from 'react';

import PropTypes from 'prop-types';

import styles from './Button.module.scss';
import { joinClassNames } from '../../../utils/utils';
import {
  ExternalLinkIconWhite,
  ExternalLinkIconRed,
  ExternalLinkIconPurple,
} from '../../../constants/icons';

const Button = (props) => {
  const {
    variant,
    type,
    href,
    label,
    onClick,
    isDisabled,
    theme,
    showExternalLinkIcon,
    containerClass,
    formik,
    ...rest
  } = props;

  // const disableButton = type === 'submit' ? !(formik.dirty && formik.isValid) : isDisabled;
  const disableButton = type === 'submit' ? !formik.isValid : isDisabled;
  const goTo = !isDisabled ? href : '';

  const linkDisabledClass = isDisabled ? styles.linkDisabled : '';
  const linkAsButtonDisabledClass = isDisabled ? styles.linkButtonDisabled : '';

  const IconForLink = theme === 'purple' ? ExternalLinkIconPurple : ExternalLinkIconRed;

  const buttonClasses =
    theme === 'purple'
      ? joinClassNames([styles.button, styles.purpleTheme, containerClass])
      : joinClassNames([styles.button, styles.redTheme, containerClass]);

  const linkClasses =
    theme === 'purple'
      ? joinClassNames([styles.link, styles.purpleTheme, linkDisabledClass, containerClass])
      : joinClassNames([styles.link, styles.redTheme, linkDisabledClass, containerClass]);

  const linkAsButtonClasses = joinClassNames([
    buttonClasses,
    styles.linkAsButton,
    linkAsButtonDisabledClass,
  ]);

  const handleClickOnLink = (e) => {
    if (isDisabled) e.preventDefault();
  };

  const renderButton = () => {
    switch (variant) {
      case 'button':
        return (
          <button className={buttonClasses} type={type} disabled={disableButton} {...rest}>
            {label}
          </button>
        );

      case 'link':
        return (
          <a
            className={linkClasses}
            href={goTo}
            target='_blank'
            onClick={handleClickOnLink}
            rel='noreferrer'
          >
            <span>{label}</span>
            {showExternalLinkIcon && (
              <img className={styles.externalLinkIcon} src={IconForLink} alt='External Link' />
            )}
          </a>
        );

      case 'link_as_button':
        return (
          <a
            className={linkAsButtonClasses}
            href={goTo}
            target='_blank'
            onClick={handleClickOnLink}
            rel='noreferrer'
          >
            {label}
            {showExternalLinkIcon && (
              <img
                className={styles.externalLinkIcon}
                src={ExternalLinkIconWhite}
                alt='External Link'
              />
            )}
          </a>
        );

      default:
        return (
          <button
            className={buttonClasses}
            type={type}
            disabled={disableButton}
            onClick={onClick}
            {...rest}
          >
            {label}{' '}
            {showExternalLinkIcon && (
              <img
                className={styles.externalLinkIcon}
                src={ExternalLinkIconWhite}
                alt='External Link'
              />
            )}
          </button>
        );
    }
  };

  return <>{renderButton()}</>;
};

Button.propTypes = {
  variant: PropTypes.oneOf(['button', 'link', 'link_as_button']),
  type: PropTypes.oneOf(['submit', 'button']),
  href: PropTypes.string,
  label: PropTypes.string,
  onClick: PropTypes.func,
  isDisabled: PropTypes.bool,
  theme: PropTypes.oneOf(['red', 'purple']),
  showExternalLinkIcon: PropTypes.bool,
  containerClass: PropTypes.string,
};

Button.defaultProps = {
  variant: 'button',
  type: 'submit',
  href: '#',
  label: '',
  onClick: () => {},
  isDisabled: false,
  theme: 'purple',
  showExternalLinkIcon: false,
  containerClass: '',
};

export default Button;
