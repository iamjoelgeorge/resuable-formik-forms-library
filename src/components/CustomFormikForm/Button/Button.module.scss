@import '../../../theme/variables.scss';
@import '../../../theme/utils.scss';

* {
  box-sizing: border-box;
}

@mixin useButtonTheme($bgColor, $hoverBgColor, $focusBgColor) {
  background-color: $bgColor;

  &:hover:not(:disabled),
  &:active {
    cursor: pointer;
    background-color: $hoverBgColor;
  }

  &:focus {
    border-color: $focusBgColor;
  }
}

@mixin useLinkTheme(
  $defaultTextColor,
  $hoverColor,
  $borderBottomColorOnFocus,
  $borderBottomColorOnActive,
  $textColorOnActive
) {
  color: $defaultTextColor;

  span {
    border: 2px solid transparent;
    padding-bottom: 2px;
  }

  &:hover:not(&.linkDisabled) {
    color: $hoverColor;
  }

  &:focus {
    span {
      border-bottom-style: dashed;
      border-bottom-color: $borderBottomColorOnFocus;
    }
  }

  &:active {
    border-color: $borderBottomColorOnActive;
    color: $textColorOnActive;
  }
}

.button,
.link {
  border: 3px solid transparent;
  border-radius: 4px;
  font-size: em(16);
  font-weight: $fontWeightExtraBold;
  line-height: 19px;
  margin-top: 12px;
  padding: 0 em(15);
  width: 198px;
  height: 51px;
  text-align: center;
}

.button {
  color: $white;
  transition: 0.1s all ease-in-out;
  outline: none;

  &.purpleTheme {
    @include useButtonTheme($primary, $darkPurple, $brightPurple);
  }

  &.redTheme {
    @include useButtonTheme($tooltipRed, $buttonHover, $pinkOne);
  }

  &:disabled {
    background-color: $white;
    color: $tertiaryGrey;
    border-color: $dGrey;
  }
}

.linkAsButton {
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;

  &.linkButtonDisabled {
    background-color: $white !important;
    cursor: not-allowed !important;
    color: $tertiaryGrey;
    border-color: $dGrey;
  }
}

.link {
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  outline: none;

  &.purpleTheme {
    @include useLinkTheme($primary, $darkPurple, $darkPurple, $brightPurple, $darkPurple);
  }

  &.redTheme {
    @include useLinkTheme($tooltipRed, $buttonHover, $tooltipRed, $pinkOne, $buttonHover);
  }

  &.linkDisabled {
    color: $tertiaryGrey;
    pointer-events: none;

    .externalLinkIcon {
      -webkit-filter: invert(50%); /* safari 6.0 - 9.0 */
      filter: invert(50%);
    }
  }
}

.externalLinkIcon {
  margin-left: em(8);
}
