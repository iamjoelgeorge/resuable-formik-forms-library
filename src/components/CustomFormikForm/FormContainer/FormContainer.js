import React, { Children, cloneElement } from 'react';
import { Formik, Form } from 'formik';
import * as Yup from 'yup';
import { validateInput } from '../../../utils/validateInput';

const FormContainer = (props) => {
  const { initialValues, children, validations, containerClass, onSubmit } = props;

  const validationObject = {};
  validations.forEach((item) => {
    validateInput(item, validationObject);
  });

  const validationSchema = Yup.object().shape(validationObject);

  return (
    <Formik initialValues={initialValues} onSubmit={onSubmit} validationSchema={validationSchema}>
      {(formik) => {
        const childrenWithExtraProp = Children.map(children, (child) =>
          cloneElement(child, { formik }),
        );
        return <Form className={containerClass}>{childrenWithExtraProp}</Form>;
      }}
    </Formik>
  );
};

export default FormContainer;
