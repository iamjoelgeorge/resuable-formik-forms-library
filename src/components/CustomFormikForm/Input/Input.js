import React, { useState, useRef } from 'react';

import PropTypes from 'prop-types';
import { Field } from 'formik';

import styles from './Input.module.scss';
import { joinClassNames } from '../../../utils/utils';
import ErrorText from '../ErrorText/ErrorText';
import SlidingLabel from '../SlidingLabel/SlidingLabel';
import Link from '../Link/Link';
import ToolTip from '../ToolTip/ToolTip';
import OptionalText from '../OptionalText/OptionalText';

const Input = (props) => {
  const {
    label,
    name,
    placeholder,
    formik,
    containerClass: customContainerClass,
    tooltipIconBoxHeading,
    tooltipIconBoxDescription,
    tooltipIconChildElement,
    tooltipLink,
    tooltipLinkText,
    helpLinkText,
    helpLink,
    optionalText,
    disabled = false,
    isRequired,
    ...rest
  } = props;
  const { values, errors } = formik;
  const inputValue = values[name];

  const [labelView, showLabelView] = useState(false);
  const [slideLabel, setSlideLabel] = useState(false);
  const inputRef = useRef();

  const userHasVisitedTheInputField = formik.touched[name];
  const inputFieldHasErrors = errors[name];
  const addErrorClassesToLabelAndInput = !!userHasVisitedTheInputField && !!inputFieldHasErrors;

  const showTooltipIcon =
    tooltipIconBoxHeading || tooltipIconBoxDescription || tooltipIconChildElement;

  const containerClasses = joinClassNames([styles.container, customContainerClass]);

  const fieldClasses = addErrorClassesToLabelAndInput
    ? joinClassNames([styles.input, styles.inputError])
    : [styles.input];

  const placeholderButtonClasses = joinClassNames([fieldClasses, styles.placeholderButton]);

  const handleFocus = () => {
    setSlideLabel(true);
  };

  const handleBlur = () => {
    setSlideLabel(false);
    if (inputValue) showLabelView(true);
  };

  const handleClick = () => {
    showLabelView(false);

    setTimeout(() => {
      inputRef?.current?.focus();
    }, 0);
  };

  const renderFieldView = () =>
    labelView && !disabled && inputValue ? (
      <button className={placeholderButtonClasses} onClick={handleClick}>
        {placeholder && !inputValue ? placeholder : inputValue}
      </button>
    ) : (
      <Field
        innerRef={inputRef}
        className={fieldClasses}
        name={name}
        id={name}
        placeholder={placeholder}
        disabled={disabled}
        onFocus={handleFocus}
        onBlur={handleBlur}
        {...rest}
      />
    );

  return (
    <div className={containerClasses}>
      <div className={styles.inputWithTooltipContainer}>
        {renderFieldView()}

        {label && (
          <SlidingLabel
            label={label}
            inputEntered={!!inputValue || !!placeholder || slideLabel}
            htmlFor={name}
            showErrorStyle={addErrorClassesToLabelAndInput}
            inputIsRequired={isRequired}
          />
        )}

        {showTooltipIcon && (
          <div className={styles.toolTipWithImage}>
            <ToolTip
              heading={tooltipIconBoxHeading}
              description={tooltipIconBoxDescription}
              descriptionElement={tooltipIconChildElement}
            />
          </div>
        )}
      </div>

      {errors[name] && formik.touched[name] && <ErrorText fieldName={name} />}

      {optionalText && (
        <OptionalText
          containerClass={styles.optionalText}
          variant='text'
          optionalText={optionalText}
        />
      )}

      {helpLinkText && (
        <OptionalText
          containerClass={styles.optionalText}
          variant='help link'
          helpLinkText={helpLinkText}
          helpLink={helpLink}
        />
      )}

      {tooltipLinkText && (
        <OptionalText
          containerClass={styles.optionalText}
          variant='tooltip link'
          tooltipLinkText={tooltipLinkText}
          tooltipLink={tooltipLink}
        />
      )}
    </div>
  );
};

Input.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  formik: PropTypes.object,
  containerClass: PropTypes.string,
  tooltipIconBoxHeading: PropTypes.string,
  tooltipIconBoxDescription: PropTypes.string,
  tooltipIconChildElement: PropTypes.element,
  tooltipLink: PropTypes.string,
  tooltipLinkText: PropTypes.string,
  helpLinkText: PropTypes.string,
  helpLink: PropTypes.string,
  optionalText: PropTypes.string,
  disabled: PropTypes.bool,
  isRequired: PropTypes.bool,
};

export default Input;
