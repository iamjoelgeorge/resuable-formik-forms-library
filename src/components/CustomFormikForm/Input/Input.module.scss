@import '../../../theme/variables.scss';
@import '../../../theme/utils.scss';

.container {
  position: relative;
  margin-top: em(30);

  .inputWithTooltipContainer {
    display: flex;
    position: relative;

    .toolTip {
      position: absolute;
      right: 0;
      margin-right: em(2);
      width: em(16);
      height: em(16);
      cursor: pointer;

      img {
        width: 100%;
        height: auto;
      }
    }
  }

  .label {
    color: $darkPurple;
    font-weight: $fontWeightSemiBold;
    position: absolute;
    top: -5px;
    left: 10px;
    transition: 0.1s all ease-in-out;
    font-size: em(14);
  }

  .slideLabel {
    top: -15px;
    font-size: em(11);
  }

  .input {
    flex: 1;
    color: $primaryDark;
    padding: em(5) em(30) em(5) 10px;
    background-color: $white;
    border: none;
    border-bottom: em(2) solid $lighterGrey;
    outline: none;
    transition: 0.1s all ease-in-out;

    &::placeholder {
      font-size: 14px;
    }

    &:placeholder-shown {
      & + label {
        @extend .slideLabel;
      }
    }

    &:hover {
      border-color: $brightPurple;
      cursor: pointer;
    }

    &:focus {
      border-color: $brightPurple;
      border-bottom-style: dashed;

      & + label {
        @extend .slideLabel;
      }
    }

    &:disabled {
      border-bottom-color: $lighterGrey;
      cursor: not-allowed;

      & + label {
        font-weight: $fontWeightRegular;
        color: $secondaryGrey;
      }
    }

    &:-webkit-autofill,
    &:-webkit-autofill:hover,
    &:-webkit-autofill:focus,
    &:-webkit-autofill:active {
      box-shadow: 0 0 0 30px $white inset !important;
      -webkit-box-shadow: 0 0 0 30px $white inset !important;
    }
  }

  .labelError {
    color: $error;
  }

  .inputError,
  .inputError:hover,
  .inputError:focus {
    border-bottom-color: $error;
    color: $secondaryGrey;
  }

  .errorContainer {
    padding-left: em(10);
    margin-top: em(5);
  }

  // .helpLinkContainer {
  //   padding-left: em(10);

  //   .helpLink {
  //     font-size: em(11);
  //     color: $secondaryGrey;
  //   }
  // }

  // .tooltipContainer {
  //   padding-left: em(10);

  //   .tooltipLink {
  //     font-size: em(11);
  //     color: $tooltipRed;
  //     text-decoration-style: dashed;
  //     transition: 0.1s all ease-in-out;
  //     outline: none;

  //     &:hover {
  //       background-color: $tooltipRed;
  //       color: $white;
  //     }

  //     &:focus {
  //       color: $pinkOne;
  //       background-color: $white;
  //     }
  //   }
  // }
}
