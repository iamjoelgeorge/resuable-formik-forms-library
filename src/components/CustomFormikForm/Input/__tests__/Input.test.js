import React from 'react';

import { render, cleanup, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import Button from '../../Button/Button';
import FormContainer from '../../FormContainer/FormContainer';
import Input from '../Input';

const defaultProps = {
  name: 'name',
  label: 'Name',
  placeholder: '',
  containerClass: '',
  tooltipBoxHeading: '',
  tooltipBoxDescription: '',
  tooltipBoxDescriptionElement: null,
  tooltipLink: '',
  tooltipLinkText: '',
  helpLinkText: '',
  helpLink: '',
  optionalText: '',
  isDisabled: false,
  isRequired: false,
  formik: {
    errors: {
      name: '',
    },
    touched: { name: false },
    values: { name: '' },
  },
};

const initialValues = {
  name: '',
};

const defaultValidations = [];
const handleSubmit = jest.fn();

afterEach(cleanup);

describe('[Component]: Input', () => {
  it('should match the Snapshot', () => {
    const tree = renderer
      .create(
        <FormContainer
          initialValues={initialValues}
          validations={defaultValidations}
          onSubmit={handleSubmit}
        >
          <Input {...defaultProps} />
        </FormContainer>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the input correctly', () => {
    const { getByTestId } = render(
      <FormContainer
        initialValues={initialValues}
        validations={defaultValidations}
        onSubmit={handleSubmit}
      >
        <Input {...defaultProps} />
      </FormContainer>
    );

    expect(getByTestId('input-name')).toBeTruthy();
  });

  it('should render a disabled input', () => {
    const props = {
      ...defaultProps,
      isDisabled: true,
    };

    const { getByTestId } = render(
      <FormContainer
        initialValues={initialValues}
        validations={defaultValidations}
        onSubmit={handleSubmit}
      >
        <Input {...props} />
      </FormContainer>
    );

    expect(getByTestId('input-name')).toHaveAttribute('disabled');
  });

  it('should render the placeholder button', () => {
    const { getByTestId } = render(
      <FormContainer
        initialValues={initialValues}
        validations={defaultValidations}
        onSubmit={handleSubmit}
      >
        <Input {...defaultProps} />
      </FormContainer>
    );

    userEvent.type(getByTestId('input-name'), 'John');
    getByTestId('input-name').blur();
    expect(getByTestId('label-view-name')).toBeTruthy();
  });

  it('should render the optional text', () => {
    const props = {
      ...defaultProps,
      optionalText: 'Test Optional text',
    };

    const { getByText } = render(
      <FormContainer
        initialValues={initialValues}
        validations={defaultValidations}
        onSubmit={handleSubmit}
      >
        <Input {...props} />
      </FormContainer>
    );

    expect(getByText('Test Optional text')).toBeTruthy();
  });

  it('should render the help link', () => {
    const props = {
      ...defaultProps,
      helpLinkText: 'Test help link',
      helpLink: 'https://virginaustralia.com/',
    };

    const { getByText } = render(
      <FormContainer
        initialValues={initialValues}
        validations={defaultValidations}
        onSubmit={handleSubmit}
      >
        <Input {...props} />
      </FormContainer>
    );

    expect(getByText('Test help link')).toBeTruthy();
  });

  it('should render the tooltip link', () => {
    const props = {
      ...defaultProps,
      tooltipLinkText: 'Test tooltip link',
      tooltipLink: 'https://virginaustralia.com/',
    };

    const { getByText } = render(
      <FormContainer
        initialValues={initialValues}
        validations={defaultValidations}
        onSubmit={handleSubmit}
      >
        <Input {...props} />
      </FormContainer>
    );

    expect(getByText('Test tooltip link')).toBeTruthy();
  });

  it('should show the error component for a required input', async () => {
    const props = {
      ...defaultProps,
      name: 'email',
      label: 'Email',
      isRequired: true,
    };

    const validations = [
      {
        name: 'email',
        type: 'email',
        isRequired: true,
        message: 'Enter the correct email id',
      },
    ];

    const initialValues = {
      email: '',
    };

    const { getByText, getByTestId } = render(
      <FormContainer
        initialValues={initialValues}
        validations={validations}
        onSubmit={handleSubmit}
      >
        <Input {...props} />
        <Button label='Submit' />
      </FormContainer>
    );

    const submitButton = getByText('Submit');
    userEvent.click(submitButton);

    await waitFor(() => {
      expect(getByTestId('error-email')).toHaveTextContent(
        'Enter the correct email id'
      );
    });
  });

  it('should show the error component for maximum characters allowed', async () => {
    const props = {
      ...defaultProps,
      name: 'name',
      label: 'name',
    };

    const validations = [
      {
        name: 'name',
        type: 'string',
        maxChars: { num: 5, message: 'You can enter only upto 5 characters' },
      },
    ];

    const initialValues = {
      name: '',
    };

    const { getByText, getByTestId } = render(
      <FormContainer
        initialValues={initialValues}
        validations={validations}
        onSubmit={handleSubmit}
      >
        <Input {...props} />
        <Button label='Submit' />
      </FormContainer>
    );

    const input = getByTestId('input-name');
    const submitButton = getByText('Submit');

    userEvent.type(input, 'Less');
    userEvent.click(submitButton);

    await waitFor(() => {
      const labelViewButton = getByTestId('label-view-name');
      userEvent.click(labelViewButton);
    });

    userEvent.clear(input);
    userEvent.type(input, 'More than 5 characters');

    await waitFor(() => {
      expect(getByTestId('error-name')).toHaveTextContent(
        'You can enter only upto 5 characters'
      );
    });
  });

  it('should show the error component for minimum characters required', async () => {
    const props = {
      ...defaultProps,
      name: 'name',
      label: 'name',
    };

    const validations = [
      {
        name: 'name',
        type: 'string',
        minChars: { num: 5, message: 'Please enter at least 5 characters' },
      },
    ];

    const initialValues = {
      name: '',
    };

    const { getByText, getByTestId } = render(
      <FormContainer
        initialValues={initialValues}
        validations={validations}
        onSubmit={handleSubmit}
      >
        <Input {...props} />
        <Button label='Submit' />
      </FormContainer>
    );

    const input = getByTestId('input-name');
    const submitButton = getByText('Submit');

    userEvent.type(input, 'More than 5 characters');
    userEvent.click(submitButton);

    await waitFor(() => {
      const labelViewButton = getByTestId('label-view-name');
      userEvent.click(labelViewButton);
    });

    userEvent.clear(input);
    userEvent.type(input, 'jn');

    await waitFor(() => {
      expect(getByTestId('error-name')).toHaveTextContent(
        'Please enter at least 5 characters'
      );
    });
  });

  it('should show the error component for minimum and maximum characters', async () => {
    const props = {
      ...defaultProps,
      name: 'name',
      label: 'name',
    };

    const validations = [
      {
        name: 'name',
        type: 'string',
        isRequired: false,
        minChars: { num: 5, message: 'Please enter at least 5 characters' },
        maxChars: { num: 10, message: 'You can enter only upto 10 characters' },
      },
    ];

    const initialValues = {
      name: '',
    };

    const { getByText, getByTestId } = render(
      <FormContainer
        initialValues={initialValues}
        validations={validations}
        onSubmit={handleSubmit}
      >
        <Input {...props} />
        <Button label='Submit' />
      </FormContainer>
    );

    const input = getByTestId('input-name');
    const submitButton = getByText('Submit');

    userEvent.type(input, 'valid input');
    userEvent.click(submitButton);

    await waitFor(() => {
      const labelViewButton = getByTestId('label-view-name');
      userEvent.click(labelViewButton);
    });

    userEvent.clear(input);
    userEvent.type(input, 'jn');

    await waitFor(() => {
      expect(getByTestId('error-name')).toHaveTextContent(
        'Please enter at least 5 characters'
      );
    });

    userEvent.clear(input);
    userEvent.type(input, 'Entering more than 10 characters');

    await waitFor(() => {
      expect(getByTestId('error-name')).toHaveTextContent(
        'You can enter only upto 10 characters'
      );
    });
  });
});
