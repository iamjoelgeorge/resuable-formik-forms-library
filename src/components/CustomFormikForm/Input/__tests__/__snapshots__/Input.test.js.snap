// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: Input should match the Snapshot 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="container "
  >
    <div
      className="inputWithTooltipContainer"
    >
      <input
        className={
          Array [
            "input",
          ]
        }
        data-testid="input-name"
        disabled={false}
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        value=""
      />
      <div
        className="container  "
      >
        <label
          className="label"
          htmlFor="name"
        >
          Name
        </label>
      </div>
    </div>
    
    
    
  </div>
</form>
`;
